lib/logstash/filters/geoip.rb

```diff
@@ -163,31 +169,24 @@ def fail_filter
                  :healthy_database => @healthy_database)
   end
 
-  def terminate_filter
-    @logger.info("geoip plugin is terminating")
-    pipeline_id = execution_context.pipeline_id
-    execution_context.agent.stop_pipeline(pipeline_id)
-  end
-
   def close
-    @database_manager.close unless @database_manager.nil?
+    @database_manager.unsubscribe_database_path(@default_database_type, self) if @database_manager
   end
 
   def select_database_path
-    vendor_path = ::File.expand_path(::File.join("..", "..", "..", "..", "vendor"), __FILE__)
-
     if load_database_manager?
-      @database_manager = LogStash::Filters::Geoip::DatabaseManager.new(self, @database, @default_database_type, vendor_path)
-      @database_manager.database_path
+      @database_manager = LogStash::Filters::Geoip::DatabaseManager.instance
+      @database_manager.subscribe_database_path(@default_database_type, @database, self)
     else
+      vendor_path = ::File.expand_path(::File.join("..", "..", "..", "..", "vendor"), __FILE__)
       @database.nil? ? ::File.join(vendor_path, "GeoLite2-#{@default_database_type}.mmdb") : @database
     end
   end
 
   def load_database_manager?
     begin
       require_relative ::File.join(LogStash::Environment::LOGSTASH_HOME, "x-pack", "lib", "filters", "geoip", "database_manager")
-      true
+      LogStash::Filters::Geoip::DatabaseManager.respond_to?(:instance)
```

Comment:
As an alternative you can check the actual logstash version with

```
irb(main):008:0> LOGSTASH_VERSION
=> "7.13.0"
irb(main):009:0> Gem::Requirement.new(">= 7.14.0").satisfied_by?(Gem::Version.new(LOGSTASH_VERSION))
=> false
```


```ruby
MINIMUM_LOGSTASH_VERSION=">= 7.14.0"

def compatible_logstash_version?
  Gem::Requirement.new(MINIMUM_LOGSTASH_VERSION).satisfied_by?(LOGSTASH_VERSION)
end
```

```suggestion
      compatible_logstash_version?
```